CMakeLists.txt

```diff
@@ -156,6 +156,12 @@ include_directories(SYSTEM ${LZ4_INCLUDE_DIR})
 add_library(lz4static STATIC IMPORTED)
 set_target_properties(lz4static PROPERTIES IMPORTED_LOCATION ${LZ4_STATIC_LIB})
 
+## ZLIB
+find_package(ZLIB REQUIRED)
+include_directories(SYSTEM ${ZLIB_INCLUDE_DIRS})
+add_library(z STATIC IMPORTED)
+set_target_properties(z PROPERTIES IMPORTED_LOCATION ${ZLIB_LIBRARIES})
```

Comment:
Also can you call the library target `zlibstatic` to be consistent with the others? I'm going to open a JIRA to improve our support for dynamic linking the dependencies